zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -292,27 +292,56 @@ where
                 data: None,
             })?;
 
-            let block_verifier_response: std::result::Result<block::Hash, VerifyBlockError> =
-                block_verifier
-                    .ready()
-                    .await
-                    .map_err(|error| Error {
-                        code: ErrorCode::ServerError(0),
-                        message: error.to_string(),
-                        data: None,
-                    })?
-                    .call(Arc::new(block))
-                    .await;
-
-            let submit_block_response = match block_verifier_response {
-                Ok(_block_hash) => submit_block::Response::Accepted,
-                Err(VerifyBlockError::Block {
-                    source: BlockError::AlreadyInChain(..),
-                }) => submit_block::Response::Duplicate,
-                Err(_) => submit_block::Response::Rejected,
+            let block_verifier_response = block_verifier
+                .ready()
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?
+                .call(Arc::new(block))
+                .await;
+
+            let chain_error = match block_verifier_response {
+                // Currently, this match arm returns `null` (Accepted) for blocks committed
+                // to any chain, but Accepted is only for blocks in the best chain.
```

Comment:
I checked each error to make sure it was being handled correctly, and added TODO comments for any differences. We can fix up any important differences when we do ticket #5487.

(There might not be any important differences.)

Reply:
The best chain note hadn't occurred to me and will be good to keep in mind.

> There might not be any important differences

Spec conformance seems particularly important in lieu of substantial user feedback.

Reply:
> Spec conformance seems particularly important in lieu of substantial user feedback.

Just to be clear, RPC error values are not part of the Zcash specification.

Only consensus-critical rules get put in the specification. Everything else can be changed by nodes, as long it is compatible with other software in the ecosystem.

There is some RPC user documentation that is auto-generated from `zcashd`. But we have found some parts of it that are outdated already. So I'm not sure if any mining pools rely on these error values.